### PR TITLE
feat: add note card

### PR DESCRIPTION
**Pull request by Letushev**
Create `AddNoteCard` component. Also, add `/add-note` route.

Screenshot 2024-05-14 at 14 27 11

### CODE REVIEW

File: src/_codux/boards/add-note-card/add-note-card.board.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createBoard } from '@wixc3/react-board';
+import { AddNoteCard } from '../../../components/add-note-card/add-note-card';
+import { ComponentWrapper } from '../../board-wrappers/component-wrapper';
+
+export default createBoard({
+    name: 'AddNoteCard',
+    Board: () => (
+        <ComponentWrapper>
```
Comment (ynesteruk) [inline src/_codux/boards/add-note-card/add-note-card.board.tsx]: What is `ComponentWrapper`?

Comment (ynesteruk) [reply src/_codux/boards/add-note-card/add-note-card.board.tsx]: Can we rename it to a FakeRouter or something like this?

File: src/components/add-note-card/add-note-card.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import classNames from 'classnames';
+import styles from './add-note-card.module.scss';
+import { Button } from '../button/button';
+import { PlusIcon } from '@radix-ui/react-icons';
+import { useNavigate, Link } from 'react-router-dom';
```
Comment (ynesteruk) [inline src/components/add-note-card/add-note-card.tsx]: Don't see any usage of `Link`

File: src/_codux/boards/add-note-card/add-note-card.board.tsx (hunk 1 above)
Comment (Letushev) [reply src/_codux/boards/add-note-card/add-note-card.board.tsx]: Agree.

Comment (ynesteruk) [reply src/_codux/boards/add-note-card/add-note-card.board.tsx]: And another point here. You need to wrap a part of a markup, with `ContentSlot` (part of '@wixc3/react-board'), to omit such fake wrapper when using it via AddPanel

File: src/components/add-note-card/add-note-card.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface AddNoteCardProps {
+    className?: string;
+}
+
+export const AddNoteCard = ({ className }: AddNoteCardProps) => {
+    const navigate = useNavigate();
+    return (
+        <div className={classNames(styles.root, className)}>
+            <Button onClick={() => navigate('/add-note')}>
```
Comment (ynesteruk) [inline src/components/add-note-card/add-note-card.tsx]: We can try to create another component.
`LinkButton` with typed `to` property as a union.

Then anyone would be able to use it via AddPanel and configure where it navigates with Props panel.

Screenshot 2024-05-14 at 14 39 20

File: src/_codux/boards/add-note-card/add-note-card.board.tsx (hunk 1 above)
Comment (ynesteruk) [reply src/_codux/boards/add-note-card/add-note-card.board.tsx]: > Agree. Maybe also add a component template with a fake router?

Sounds good. But i'm not sure it's a component or board template

File: src/components/add-note-card/add-note-card.tsx (hunk 3 above)
Comment (Letushev) [reply src/components/add-note-card/add-note-card.tsx]: But `to` will be dynamic in some cases, like a page to edit a note, `/notes/:id` somethings like this. So I can't make it a union.

**Review by ynesteruk — APPROVED** [review]

**Pull request merged**